src/shared/destinations.ts

```diff
@@ -114,6 +114,24 @@ const destinationConfigs: DestinationConfig[] = [
       return `${repoUrl}/blob/HEAD/CHANGELOG.md`;
     },
   },
+  {
+    keywords: ["d"],
+    generateUrl: async (packageName) => {
+      let repoUrl = await getRepoUrl(packageName, {
+        skipDirectoryTrimming: true,
+      });
+
+      if (repoUrl?.includes("://github.com")) {
+        repoUrl = repoUrl.replace("://github.com", "://github.dev");
+        const packageMetadata = await getPackageMetadata(packageName);
+        const { directory } = packageMetadata.repository as JsonObject;
+
+        return directory ? `${repoUrl}/tree/main/${directory}/` : repoUrl;
+      }
+
+      return repoUrl;
```

Comment:
For GitLab or other services will just redirect to repository? 🤔 @kachkaev

Reply:
Yeah that’s a good point. The rule I try to stick with is redirecting people as close as possible to the destination that they’ve picked. E.g. because there are no releases on GitLab, we redirect to tags. The same would apply here.

Reply:
Actually I just remembered that they’ve got their own [Web IDE](https://docs.gitlab.com/ee/user/project/web_ide/) 👀

Check it out:
https://gitlab.com/gitlab-org/gitlab/
           ↓
<img width="120" alt="Screenshot 2021-08-19 at 23 34 58" src="https://user-images.githubusercontent.com/608862/130153053-06a16838-a78e-44d6-b9c6-44aedfb48c5f.png">  
           ↓
https://gitlab.com/-/ide/project/gitlab-org/gitlab/edit/

Maybe we could redirect folks there?

Reply:
Removed subdirectory support, because it seems is not consistent